actors/migration/nv16/top.go

```diff
@@ -66,38 +66,62 @@ func ActorHeadKey(addr address.Address, head cid.Cid) string {
+func MigrateStateTree(ctx context.Context, store cbor.IpldStore, actorsManifest cid.Cid, actorsRootIn cid.Cid, priorEpoch abi.ChainEpoch, cfg Config, log Logger, cache MigrationCache) (cid.Cid, error) {
 	if cfg.MaxWorkers <= 0 {
 		return cid.Undef, xerrors.Errorf("invalid migration config with %d workers", cfg.MaxWorkers)
 	}
 
+	adtStore := adt8.WrapStore(ctx, store)
+
+	// load manifest
+	var manifest manifest8.Manifest
+	if err := adtStore.Get(ctx, actorsManifest, &manifest); err != nil {
+		return cid.Undef, xerrors.Errorf("error reading actor manifest: %w", err)
+	}
+
+	if err := manifest.Load(ctx, adtStore); err != nil {
+		return cid.Undef, xerrors.Errorf("error loading actor manifest: %w", err)
+	}
+
 	// Maps prior version code CIDs to migration functions.
-	var migrations = map[cid.Cid]actorMigration{
-		builtin7.AccountActorCodeID:          nilMigrator{builtin8.AccountActorCodeID},
-		builtin7.CronActorCodeID:             nilMigrator{builtin8.CronActorCodeID},
-		builtin7.InitActorCodeID:             nilMigrator{builtin8.InitActorCodeID},
-		builtin7.MultisigActorCodeID:         nilMigrator{builtin8.MultisigActorCodeID},
-		builtin7.PaymentChannelActorCodeID:   nilMigrator{builtin8.PaymentChannelActorCodeID},
-		builtin7.RewardActorCodeID:           nilMigrator{builtin8.RewardActorCodeID},
-		builtin7.StorageMarketActorCodeID:    nilMigrator{builtin8.StorageMarketActorCodeID},
-		builtin7.StorageMinerActorCodeID:     nilMigrator{builtin8.StorageMinerActorCodeID},
-		builtin7.StoragePowerActorCodeID:     nilMigrator{builtin8.StoragePowerActorCodeID},
-		builtin7.SystemActorCodeID:           nilMigrator{builtin8.SystemActorCodeID},
-		builtin7.VerifiedRegistryActorCodeID: nilMigrator{builtin8.VerifiedRegistryActorCodeID},
+	migrations := make(map[cid.Cid]actorMigration)
+
+	// simple code migrations
+	var simpleMigrations = map[string]cid.Cid{
+		"init":             builtin7.InitActorCodeID,
+		"cron":             builtin7.CronActorCodeID,
+		"account":          builtin7.AccountActorCodeID,
+		"storagepower":     builtin7.StoragePowerActorCodeID,
+		"storageminer":     builtin7.StorageMinerActorCodeID,
+		"storagemarket":    builtin7.StorageMarketActorCodeID,
+		"paymentchannel":   builtin7.PaymentChannelActorCodeID,
+		"multisig":         builtin7.MultisigActorCodeID,
+		"reward":           builtin7.RewardActorCodeID,
+		"verifiedregistry": builtin7.VerifiedRegistryActorCodeID,
+	}
+
+	for name, code7Cid := range simpleMigrations { //nolint:nomaprange
+		code8Cid, ok := manifest.Get(name)
+		if !ok {
+			return cid.Undef, xerrors.Errorf("code cid for %s actor not found in manifest", name)
+		}
+
+		migrations[code7Cid] = codeMigrator{code8Cid}
 	}
 
-	// Set of prior version code CIDs for actors to defer during iteration, for explicit migration afterwards.
-	var deferredCodeIDs = map[cid.Cid]struct{}{
```

Comment:
Don't remue this please its been helpful in the past and nice to keep easily available for future migrations.

Reply:
ok, i will keep it.